File: archive/aws-intro/csharp/lab-01/code/05-making-your-stack-configurable/step4.cs
using System.IO;
using Pulumi;
using Pulumi.Aws.S3;
using Pulumi.Aws.S3.Inputs;
using Aws = Pulumi.Aws;

class MyStack : Stack
{
    public MyStack()
    {
        var config = new Pulumi.Config();
        var siteDir = config.Get("siteDir");
        
        var bucket = new Aws.S3.Bucket("my-bucket", new BucketArgs
        {
            Website = new BucketWebsiteArgs
            {
                IndexDocument = "index.html",
            }
        });
        this.BucketName = bucket.BucketName;
        this.WebsiteEndpoint = bucket.WebsiteEndpoint;

        foreach (var file in Directory.EnumerateFiles(siteDir))
        {
            var name = Path.GetFileName(file);
            var bucketObject = new Aws.S3.BucketObject(name, new BucketObjectArgs
            {
                Bucket = bucket.BucketName,
                Source = new FileAsset(Path.Combine(siteDir, Path.GetFileName(file))),
                Acl = "public-read",
                ContentType = "text/html",
            });
        }
    }

    [Output] public Output<string> BucketName { get; set; }
    [Output] public Output<string> WebsiteEndpoint { get; set; }
}

File: archive/aws-intro/python/lab-01/code/05-making-your-stack-configurable/step4.py
import pulumi
import pulumi_aws as aws
import os
import mimetypes

config = pulumi.Config()
site_dir = config.require("siteDir")

bucket = aws.s3.Bucket("my-bucket",
   website={
       "index_document": "index.html"
})

for file in os.listdir(site_dir):
    filepath = os.path.join(site_dir, file)
    mime_type, _ = mimetypes.guess_type(filepath)
    obj = aws.s3.BucketObject(file,
          bucket=bucket.bucket,
          source=pulumi.FileAsset(filepath),
          acl="public-read",
          content_type=mime_type
    )

pulumi.export('bucket_name', bucket.bucket)
pulumi.export('bucket_endpoint', pulumi.Output.concat("http://", bucket.website_endpoint))
